sycl/source/backend/opencl.cpp

```diff
@@ -44,6 +45,74 @@ __SYCL_EXPORT queue make_queue(const context &Context,
   return detail::make_queue(NativeHandle, Context, nullptr, false,
                             ContextImpl->get_async_handler(), backend::opencl);
 }
+
+// Free functions to query OpenCL backend extensions
+// TODO: Extensions have been deprecated for aspects
+__SYCL_EXPORT bool has_extension(const sycl::platform &SyclPlatform,
+                                 const std::string &Extension) {
+  if (SyclPlatform.get_backend() != sycl::backend::opencl) {
+    throw sycl::exception(
+        errc::backend_mismatch,
+        "has_extension can only be used with an OpenCL backend");
+  }
+
+  auto PlatformImpl = getSyclObjImpl(SyclPlatform);
+  RT::PiPlatform PluginPlatform = PlatformImpl->getHandleRef();
+  plugin Plugin = PlatformImpl->getPlugin();
+
+  // Manual invocation of plugin API to avoid using deprecated
+  // info::platform::extensions call
+  size_t ResultSize = 0;
+  Plugin.call<PiApiKind::piPlatformGetInfo>(
+      PluginPlatform, PI_PLATFORM_INFO_EXTENSIONS, 0, nullptr, &ResultSize);
+  if (ResultSize == 0) {
+    return false;
+  }
+  std::unique_ptr<char[]> Result(new char[ResultSize]);
+  Plugin.call<PiApiKind::piPlatformGetInfo>(PluginPlatform,
+                                            PI_PLATFORM_INFO_EXTENSIONS,
+                                            ResultSize, Result.get(), nullptr);
+
+  std::string ExtensionsString = Result.get();
+  std::vector<std::string> ExtensionsList = split_string(ExtensionsString, ' ');
+
+  auto FindResult =
+      std::find(ExtensionsList.begin(), ExtensionsList.end(), Extension);
+  return FindResult != ExtensionsList.end();
```

Comment:
Can't we just search Extension in the original `std::string` like this?:
```suggestion
  return ExtensionsList.find(Extension) != std::string::npos;
```